pandas/core/dtypes/common.py

```diff
@@ -1626,7 +1630,10 @@ def is_bool_dtype(arr_or_dtype):
         # guess this
         return (arr_or_dtype.is_object and
                 arr_or_dtype.inferred_type == 'boolean')
-
+    elif isinstance(arr_or_dtype, SparseDtype):
```

Comment:
what is this for?

Reply:
This is for `is_bool_indexe(SparseArray)` We support boolean indexing a SparseArray with a SparseArray of booleans.